packages/react-sdk/src/Provider.tsx

```diff
@@ -36,32 +43,83 @@ export class OptimizelyProvider extends React.Component<
   OptimizelyProviderProps,
   OptimizelyProviderState
 > {
-  sdkWrapper: UserWrappedOptimizelySDK
-
   constructor(props: OptimizelyProviderProps) {
     super(props)
+    const { optimizely, userId, userAttributes, user} = props
+
+    // check if user id/attributes are provided as props and set them ReactSDKClient
+    let finalUser: {
+      id: string
+      attributes: optimizely.UserAttributes
+    } | null = null
+
+    if (user) {
+      if ('then' in user) {
+        user.then(user => {
+          optimizely.setUser(user)
+        })
+      } else {
+        finalUser = {
+          id: user.id,
+          attributes: user.attributes || {},
+        }
+      }
+    } else if (userId) {
+      finalUser = {
+        id: userId,
+        attributes: userAttributes || {},
+      }
+      // deprecation warning
+      console.warn(
+        'Passing userId and userAttributes as props is deprecated, please switch to using `user` prop',
+      )
+    }
 
-    const { optimizely, userId, userAttributes } = props
-    this.sdkWrapper = createUserWrapper({
-      instance: optimizely,
-      userId,
-      userAttributes,
-    })
+    if (finalUser) {
+      optimizely.setUser(finalUser)
+    }
+  }
+
+  componentDidUpdate(prevProps: OptimizelyProviderProps): void {
+    if (prevProps.isServerSide) {
+      // dont react to updates on server
+      return
+    }
+    const { optimizely  } = this.props
```

Comment:
nit: there are two spaces after `optimizely`

Reply:
nit: This is still here!

Reply:
Oops, will fix this.